File: app/src/main/java/com/lucagiorgetti/surprix/model/Surprise.java
package com.lucagiorgetti.surprix.model;

import androidx.annotation.Nullable;

/**
 * Created by Utente on 17/04/2017.
 */

public class Surprise implements Comparable<Surprise>{
    private String id = null;
    private String description = null;
    private String img_path = null;
    private String code = null;
    private String set_name = null;
    private int set_year = -1;
    private String set_producer_name = null;
    private String set_product_name = null;
    private String set_producer_color = null;
    private String set_nation = null;
    private String set_id = null;
    private String rarity = null;

    private boolean set_effective_code = true;

    public Surprise(){
    }

    public Surprise(String description, String img_path, String code, Set set, @Nullable Integer rarity) {
        this.description = description;
        this.img_path = img_path;
        this.code = code;
        this.set_name = set.getName();
        this.set_year = set.getYear();
        this.set_producer_color = set.getProducer_color();
        this.set_producer_name = set.getProducer_name();
        this.set_product_name = set.getProduct();
        this.set_effective_code = set.hasEffectiveCode();
        this.set_nation = set.getNation();
        this.id = set_producer_name + "_" + set_year + "_" + set.getCode() + "_" + code;
        this.set_id = set.getId();
        this.rarity = rarity != null ? String.valueOf(rarity) : null;
        this.set_effective_code = set.hasEffectiveCode();
    }


    public String getId() {
        return id;
    }

    public String getDescription() {
        return description;
    }

    public String getImg_path() {
        return img_path;
    }

    public String getCode() {
        return code;
    }

    public String getSet_name() {
        return set_name;
    }

    public int getSet_year() {
        return set_year;
    }

    public String getSet_producer_name() {
        return set_producer_name;
    }

    public String getSet_product_name() {
        return set_product_name != null ? set_product_name : "";
    }

    public String getSet_nation() {
        return set_nation;
    }

    public String getSet_id() {
        return set_id;
    }

    public String getSet_producer_color() {
        return set_producer_color;
    }

    public String getRarity(){
        return rarity;
    }

    public boolean isSet_effective_code() {
        return set_effective_code;
    }

    public void setSet_effective_code(boolean set_effective_code) {
        this.set_effective_code = set_effective_code;
    }

    public Integer getIntRarity(){
        return rarity != null ? Integer.parseInt(this.getRarity()) : null;
    }

    public boolean has_set_effective_code() {
        return set_effective_code;
    }

    @Override
    public int compareTo(Surprise surprise) {
        try {
            int a = Integer.parseInt(this.getCode());
            int b = Integer.parseInt(surprise.getCode());
            return a - b;
        } catch (Exception ignored) {

        }
        return getCode().compareTo(surprise.getCode());
    }
}

File: app/src/main/java/com/lucagiorgetti/surprix/model/Surprise.kt
package com.lucagiorgetti.surprix.model

/**
 * Created by Utente on 17/04/2017.
 */
class Surprise : Comparable<Surprise> {
    var id: String? = null
        private set
    var description: String? = null
        private set
    var img_path: String? = null
        private set
    var code: String? = null
        private set
    var set_name: String? = null
    var set_year_year = -1
    var set_year_id: String? = null
    var set_year_name: String? = null
    var set_producer_name: String? = null
    var set_producer_id: String? = null
    var set_producer_color: String? = null
    var set_nation: String? = null
    var set_category: String? = null
    var set_id: String? = null
        private set
    var rarity: String? = null
        private set
    var isSet_effective_code = true
        private set

    constructor()
    constructor(description: String?, img_path: String?, code: String, set: Set, rarity: Int?) {
        this.description = description
        this.img_path = img_path
        this.code = code
        set_name = set.name
        set_producer_color = set.producer_color
        set_producer_name = set.producer_name
        set_producer_id = set.producer_id
        isSet_effective_code = set.isEffectiveCode
        set_category = set.category
        set_nation = set.nation
        set_year_year = set.year_year
        set_year_id = set.year_id
        set_year_name = set.year_desc
        id = set_producer_name + "_" + set_year_year + "_" + set.code + "_" + code
        set_id = set.id
        this.rarity = rarity?.toString()
    }

    val intRarity: Int?
        get() = if (rarity != null) rarity!!.toInt() else null

    override fun compareTo(surprise: Surprise): Int {
        try {
            val a = this.code!!.toInt()
            val b = surprise.code!!.toInt()
            return a - b
        } catch (ignored: Exception) {
        }
        return this.code!!.compareTo(surprise.code!!)
    }
}
